packages/form-js-viewer/src/render/components/util/valuesUtil.js

```diff
@@ -0,0 +1,49 @@
+import { get } from 'min-dash';
+
+// parses the options data from the provided form field and form data
+export function getValuesData(formField, formData) {
+  const { valuesKey, values } = formField;
+  return valuesKey ? get(formData, [ valuesKey ]) : values;
+}
+
+// transforms the provided options into a normalized format, trimming invalid options
+export function normalizeValuesData(valuesData) {
+  return valuesData.filter(_isValueSomething).map(v => _normalizeValueData(v)).filter(v => v);
+}
+
+function _normalizeValueData(valueData) {
+
+  if (_isAllowedValue(valueData)) {
+
+    // if a primitive is provided, use it as label and value
+    return { value: valueData, label: valueData };
+  }
+
+  if (typeof (valueData) === 'object') {
+    if (!valueData.label && _isAllowedValue(valueData.value)) {
```

Comment:
Quick question: Why don't we also allow for

```json
[
  { "label": "foo" }
]
```

and use the label as value as well?

Reply:
I guess the simplest way I think about it:

It's okay to derive metadata from state data
It's wrong to derive state data from metadata

Reply:
> I guess the simplest way I think about it:
> 
> It's okay to derive metadata from state data It's wrong to derive state data from metadata

💯 nice thinking, I agree 👍